File: backend/ent/schema/lesson.go
package schema

import (
	"entgo.io/ent"
	"entgo.io/ent/dialect/entsql"
	"entgo.io/ent/schema/edge"
	"entgo.io/ent/schema/field"
	"github.com/google/uuid"
)

// Lesson holds the schema definition for the Lesson entity.
type Lesson struct {
	ent.Schema
}

// Fields of the Lesson.
func (Lesson) Fields() []ent.Field {
	return []ent.Field{
		field.UUID("id", uuid.UUID{}).
			Annotations(entsql.DefaultExpr("uuid_generate_v4()")),
		field.String("title"),
		field.String("description"),
		field.String("layout"),
		field.UUID("category_id", uuid.UUID{}).Optional(),
	}
}

// Edges of the Lesson.
func (Lesson) Edges() []ent.Edge {
	return []ent.Edge{
		edge.To("completed_lessons", CompletedLesson.Type).
			Annotations(entsql.OnDelete(entsql.Cascade)),

		edge.From("lesson", Category.Type).
			Ref("lessons").
			Unique().
			Field("category_id"),
	}
}

File: frontend/src/shared/validation/lesson.ts
import { object, string } from "yup";

export const lessonSchema = object({
  title: string().trim().required("Поле Наименование обязательно для заполнения"),
  description: string().trim().required("Поле Описание обязательно для заполнения"),
  category_id: string().required("Поле категория обязательна для заполнения"),
  layout: string().trim().required("Поле Разметка обязательно для заполнения"),
});
